### PR TITLE
Main segment

### PR DESCRIPTION
**Pull request by marijnvdwerf**
This one is still a bit of a mess. Mostly PRing to show progress and see what clang-format says.

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 1139 functions (+7.74%) on `pal`.
src/effects.c:16: warning: redundant redeclaration of `gCurrentLanguage' in same scope
include/variables.h:267: warning: previous declaration of `gCurrentLanguage'
src/state_title_screen.c:100: warning: redundant redeclaration of `gCurrentLanguage' in same scope
include/variables.h:267: warning: previous declaration of `gCurrentLanguage'

File: src/os/nusys/nucontmgr.c [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+#else
+#define DO_INLINE inline
+#define DO_STATIC static
+#endif
+
+DO_STATIC s32           contRetrace(NUSiCommonMesg* mesg);
+DO_STATIC s32           contRead(NUSiCommonMesg* mesg);
+DO_STATIC s32           contReadNW(NUSiCommonMesg* mesg);
+DO_STATIC DO_INLINE s32 contReadData(OSContPad* pad, u32 lockflag);
+DO_STATIC s32           contQuery(NUSiCommonMesg* mesg);
```
Comment (HailSanta) [inline src/os/nusys/nucontmgr.c:20]: Think I'd rather see these a #if VERSION_PAL / #else block than defines involving language keywords

File: src/state_pause.c [hunk 2]
```diff
@@ -38,6 +38,27 @@ NUPiOverlaySegment D_8007795C = {
...
+    (void*)0x00169350, (void*)0x0016B3D0, (void*)0x8027AA10, (void*)0x8027AA10, (void*)0x8027AA10,
+    (void*)0x8027AA10, (void*)0x8027CA90, (void*)0x8027CA90, (void*)0x8027CA90,
+};
+NUPiOverlaySegment D_8007795C_2 = {
+    (void*)0x0016B3D0, (void*)0x0016D450, (void*)0x8027AA10, (void*)0x8027AA10, (void*)0x8027AA10,
+    (void*)0x8027AA10, (void*)0x8027CA90, (void*)0x8027CA90, (void*)0x8027CA90,
+};
+NUPiOverlaySegment D_8007795C_3 = {
+    (void*)0x0016D450, (void*)0x0016F4D0, (void*)0x8027AA10, (void*)0x8027AA10, (void*)0x8027AA10,
+    (void*)0x8027AA10, (void*)0x8027CA90, (void*)0x8027CA90, (void*)0x8027CA90,
```
Comment (HailSanta) [inline src/state_pause.c:57]: Raw pointers need to be symbolized?

File: src/item_entity_scripts.inc.c [hunk 3]
```diff
@@ -782,393 +782,395 @@ ItemScript IES_ITEM_136 = STANDARD_ITEM_SCRIPT(battle_136);
...
-    IES_Jump,
-    IES_Jump,
-    IES_Jump,
-    IES_Jump,
-    IES_Jump,
-    IES_Jump,
-    IES_Jump
-};
+s32* gItemEntityScripts[] = { IES_Jump,
+                              IES_GearBoots1,
```
Comment (nanaian) [inline src/item_entity_scripts.inc.c:786]: Need to fix clang-format so it doesn't do this

**Comment by ethteck** [comment]
@marijnvdwerf do you plan on continuing to work on this? If not, that's fine, but it might be better to close the PR and leave it as a branch on the repo (we can push to pmret:pal-main2)

**Comment by marijnvdwerf** [comment]
I think a branch might make more sense, as I don't see myself finishing this in the short term.

I mainly pushed this as a means to test the clang-format config. But that has kinda stagnated. Most work for this will be in getting the libultra functions cleaned up, and figure out how the compiler reordered stuff.

**Pull request closed without merging**